rest/api_test.go

```diff
@@ -3671,6 +3671,50 @@ func TestUnsupportedConfig(t *testing.T) {
 	sc.Close()
 }
 
+func TestImportingPurgedDocument(t *testing.T) {
+	var rt RestTester
+	defer rt.Close()
+
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("This test won't work under walrus until https://github.com/couchbase/sync_gateway/issues/2390")
+	}
+
+	if !base.TestUseXattrs() {
+		t.Skip("XATTR based tests not enabled.  Enable via SG_TEST_USE_XATTRS=true environment variable")
+	}
+
+	body := `{"_purged": true, "foo": "bar"}`
+	rt.Bucket().Add("key", 0, []byte(body))
+
+	numErrors, err := strconv.Atoi(base.StatsResourceUtilization().Get(base.StatKeyErrorCount).String())
+	assert.NoError(t, err)
+
+	response := rt.SendRequest("GET", "/db/key", "")
+	fmt.Println(response.Body)
+
+	numErrorsAfter, err := strconv.Atoi(base.StatsResourceUtilization().Get(base.StatKeyErrorCount).String())
+	assert.NoError(t, err)
+
+	assert.Equal(t, numErrors, numErrorsAfter)
+}
+
+func TestRestSettingPurged(t *testing.T) {
```

Comment:
I'm not sure what scenario this test is covering.  I think the PUT on line 3708 is going to return an error (fields with leading underscore are rejected), so the doc wouldn't be written in the first place.  If you think it's needed for a specific scenario, can you share the details?

Reply:
> I'm not sure what scenario this test is covering. I think the PUT on line 3708 is going to return an error (fields with leading underscore are rejected), so the doc wouldn't be written in the first place. If you think it's needed for a specific scenario, can you share the details?



Yes exactly, this is what is was meant to test. I was altering the way this worked as the error is the same as the one that the QE test was expecting. Added this to ensure I didn't break this original behaviour. I have however altered the test a little now to be more appropriate.